llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -5213,13 +5213,27 @@ static SDValue lowerShuffleViaVRegSplitting(ShuffleVectorSDNode *SVN,
     for (unsigned I : seq<unsigned>(Data.size())) {
       const auto &[Idx1, Idx2, _] = Data[I];
       if (Values.contains(Idx1)) {
-        assert(Idx2 != UINT_MAX && Values.contains(Idx2) &&
-               "Expected both indices to be extracted already.");
-        break;
+        // If the shuffle contains permutation of odd number of elements,
+        // Idx1 might be used already in the first iteration.
+        //
+        // Idx1 = shuffle Idx1, Idx2
+        // Idx1 = shuffle Idx1, Idx3
+        if (const auto It = Values.find(Idx1);
```

Comment:
The It variable is only used in an assert, not the condition.  Needs to either be folded into the assert, or #ifndef NDEBUG guards added to prevent non-assertions enabled build break.

Hm, looking at the check this is used for, it's entirely redundant with the contains check above.  I think you can just remove this,